### PR TITLE
ER simulations as a special case of SBM

### LINKED ISSUE #144: er simulations should be a special case of sbm
No reason to have that redundant code
Also would enable DC-ER cause why not

### LINKED ISSUE #198: fix test_omni_matrix_random()
```
def test_omni_matrix_random():
    expected_output = np.array(
        [
            [0.0, 1.0, 1.0, 0.0, 0.5, 0.5],
            [1.0, 0.0, 1.0, 0.5, 0.0, 1.0],
            [1.0, 1.0, 0.0, 0.5, 1.0, 0.0],
            [0.0, 0.5, 0.5, 0.0, 0.0, 0.0],
            [0.5, 0.0, 1.0, 0.0, 0.0, 1.0],
            [0.5, 1.0, 0.0, 0.0, 1.0, 0.0],
        ]
    )

    np.random.seed(4)
    graphs = [er_np(3, 0.5) for _ in range(2)]

    A = _get_omni_matrix(graphs)
    assert_allclose(A, expected_output)
```

Though the random seed is set this test is dependent on the implementation of ```er_np``` function, changing that implementation breaks this test. Should be rewritten to not use ```er_np``` unless I am missing something @j1c ?

### PR DESCRIPTION
**Pull request by Kikiwink**
Reference Issues/PRs
Fixes #144 
Fixes #198  

What does this implement/fix? Explain your changes.
Part of ER simulations is replaced by SBM
Modifying the omni test

Any other comments?

### CODE REVIEW

File: graspy/simulations/simulations.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def sample_edges(P, directed=False, loops=False):
         return A - np.diag(np.diag(A))
 
 
-def er_np(n, p, directed=False, loops=False, wt=1, wtargs=None):
+def er_np(n, p, directed=False, loops=False, wt=1, wtargs=None, dc=None, dc_kws={}):
```
Comment (bdpedigo) [inline graspy/simulations/simulations.py]: add documentation for the new kwargs

Comment (bdpedigo) [reply graspy/simulations/simulations.py]: add tests for the new kwargs

Comment (j1c) [reply graspy/simulations/simulations.py]: @bdpedigo I dont think there should be tests when the underlying function already does the tests.

File: graspy/simulations/simulations.py [hunk 2]
```diff
@@ -96,50 +96,46 @@ def er_np(n, p, directed=False, loops=False, wt=1, wtargs=None):
     wtargs: dictionary, optional (default=None)
         Optional arguments for parameters that can be passed
         to weight function ``wt``.
+    dc: function or array-like, shape (n_vertices)
+        `dc` is used to generate a degree-corrected stochastic block model [1] in
```
Comment (bdpedigo) [inline graspy/simulations/simulations.py]: update name (ER not SBM)

File: graspy/simulations/simulations.py [hunk 3]
```diff
@@ -96,50 +96,46 @@ def er_np(n, p, directed=False, loops=False, wt=1, wtargs=None):
...
+        which each node in the graph has a parameter to specify its expected degree
+        relative to other nodes within its community.
+
+        - function:
+            should generate a non-negative number to be used as a degree correction to
+            create a heterogenous degree distribution. A weight will be generated for
+            each vertex, normalized so that the sum of weights in each block is 1.
```
Comment (bdpedigo) [inline graspy/simulations/simulations.py]: remove "in each block"

File: graspy/simulations/simulations.py [hunk 4]
```diff
@@ -96,50 +96,46 @@ def er_np(n, p, directed=False, loops=False, wt=1, wtargs=None):
...
+        - array-like of scalars, shape (n_vertices):
+            The weights in each block should sum to 1; otherwise, they will be
```
Comment (bdpedigo) [inline graspy/simulations/simulations.py]: Remove "in each block"

File: graspy/simulations/simulations.py [hunk 5]
```diff
@@ -96,50 +96,46 @@ def er_np(n, p, directed=False, loops=False, wt=1, wtargs=None):
...
+            normalized and a warning will be thrown. The scalar associated with each
+            vertex is the node's relative expected degree within its community.
```
Comment (bdpedigo) [inline graspy/simulations/simulations.py]: Remove "within its expected community"

File: graspy/simulations/simulations.py [hunk 6]
```diff
@@ -96,50 +96,46 @@ def er_np(n, p, directed=False, loops=False, wt=1, wtargs=None):
...
-    elif p < 0:
-        msg = "You have passed a probability, {}, less than 0."
-        msg = msg.format(float(p))
-        raise ValueError(msg)
-    elif p > 1:
-        msg = "You have passed a probability, {}, greater than 1."
-        msg = msg.format(float(p))
-        raise ValueError(msg)
+    if isinstance(dc, (list, np.ndarray)) and all(callable(f) for f in dc):
+        raise TypeError("dc is not of type function or list")
```
Comment (bdpedigo) [inline graspy/simulations/simulations.py]: dc should be a list or array of scalars, or a function

File: graspy/simulations/simulations.py [hunk 7]
```diff
@@ -96,50 +96,46 @@ def er_np(n, p, directed=False, loops=False, wt=1, wtargs=None):
...
     if not np.issubdtype(type(n), np.integer):
         raise TypeError("n is not of type int.")
-    elif n <= 0:
-        msg = "n must be > 0."
-        raise ValueError(msg)
-    if type(directed) is not bool:
-        raise TypeError("directed is not of type bool.")
+    if not np.issubdtype(type(p), np.floating):
```
Comment (bdpedigo) [inline graspy/simulations/simulations.py]: add simple check for this one

**Pull request merged**